src/components/select/simple-select/simple-select.component.js

```diff
@@ -0,0 +1,363 @@
+        onChange(createCustomEvent(match.props.value));
+      }
+
+      if (isControlled.current) {
+        return previousValue;
+      }
+
+      setTextValue(match.props.text);
+
+      return match.props.value;
+    });
+  }, [children, createCustomEvent, onChange]);
+
+  const handleTextboxChange = useCallback((event) => {
+    const newValue = event.target.value;
+    const newCharacter = newValue.slice(-1);
+
+    if (isTimerCounting.current) {
+      const newVal = filterText.current + newCharacter;
+
+      filterText.current = newVal;
+      selectValueStartingWithText(newVal);
+      clearTimeout(filterTimer.current);
+    } else {
+      filterText.current = newCharacter;
+      selectValueStartingWithText(newCharacter);
+    }
+
+    isTimerCounting.current = true;
+    clearTimeout(filterTimer.current);
+
+    filterTimer.current = setTimeout(() => {
+      isTimerCounting.current = false;
+      filterText.current = '';
+    }, 500);
+  }, [selectValueStartingWithText]);
+
+  const handleTextboxKeydown = useCallback(((event) => {
+    const keyCode = event.which;
+
+    if (isNavigationKey(event)) {
+      setSelectedValue((previousSelectedValue) => {
+        const nextElement = getNextChildByKeyCode(keyCode, children, previousSelectedValue);
+
+        setTextValue(nextElement.props.text);
+
+        return nextElement.props.value;
+      });
+    }
+
+    if (Events.isEnterKey(event) || Events.isSpaceKey(event)
+      || isNavigationKey(event)) {
+      event.preventDefault();
+      setOpen(true);
+    }
+  }), [children, setOpen]);
+
+  const handleGlobalClick = useCallback((event) => {
+    const notInContainer = containerRef.current && !containerRef.current.contains(event.target);
+    const notInList = listboxRef.current && !listboxRef.current.contains(event.target);
+
+    if (notInContainer && notInList) {
+      setMatchingText(selectedValue);
+      setOpen(false);
+      filterText.current = '';
+    }
+  }, [setMatchingText, setOpen, selectedValue]);
+
+  useEffect(() => {
+    const newValue = value || defaultValue;
+    const message = 'Input elements should not switch from uncontrolled to controlled (or vice versa). '
+    + 'Decide between using a controlled or uncontrolled input element for the lifetime of the component';
+    invariant(isControlled.current === (value !== undefined), message);
+
+    setSelectedValue(newValue);
+    setMatchingText(newValue);
+  }, [value, defaultValue, setMatchingText]);
+
+  useEffect(() => {
+    const clickEvent = 'click';
+
+    document.addEventListener(clickEvent, handleGlobalClick);
+
+    return function cleanup() {
+      document.removeEventListener(clickEvent, handleGlobalClick);
+    };
+  }, [handleGlobalClick]);
+
+  useEffect(() => {
+    return function cleanup() {
+      clearTimeout(filterTimer.current);
+    };
+  }, []);
+
+  function handleTextboxClick(event) {
+    if (onClick) {
+      onClick(event);
+    }
+
+    setOpenState((isAlreadyOpen) => {
```

Comment:
I think you need to add a guard here `!openOnFocus` or you might toggle the open state inadvertently when `handleTextboxFocus` is called

Reply:
good point, added